### DISCUSSION TITLE
How do I resolve l5d-client-id?

### DISCUSSION THREAD

**Discussion by ruz** [linkerd/linkerd2#9835] [Q&A]
Linkerd's proxy adds this header to incoming requests. How can I convert it to pod ip? Not by hands, via API.

**Comment by kleimkuhler** [comment]
Unfortunately there is not a way to do this right now. This is tracked in #4219 though and there has some been recent activity, so it's on our radar. I'd subscribe to that issue so that you're updated with any progress on it.

**Comment by ruz** [comment]
Placed thumb up and subscribed to #4219.

I've found code where `l5d-client-id` header [was added](https://github.com/linkerd/linkerd2-proxy/pull/981/files). First time reading Rust. Looks like this id is a property of the TLS connection. Probably is set by sender or set on receiving end. Hardly believe it's random. Can you point to code that creates the value?

What was the point in adding this header? How can I use it if I can not map it to any entities in the cluster?

**Reply by hawkw** [reply]
The "id" in the name of this header refers to the client's _identity_, rather than an _identifier_. The header contains the name of the Kubernetes service account that the client's mTLS client cert was issued for. It can't be used to look up a pod IP or similar, because a number of pods may be part of the same service account. Instead, it simply indicates what service account the client workload is part of.
